src/main.js

```diff
@@ -1,111 +1,148 @@
-              <button>공유</button>
-            </div>
-          </div>
-
-          <div class="bg-white rounded-lg shadow p-4">
-            <div class="flex items-center mb-2">
-              <img src="https://via.placeholder.com/40" alt="프로필" class="rounded-full mr-2">
-              <div>
-                <p class="font-bold">정수연</p>
-                <p class="text-sm text-gray-500">2시간 전</p>
-              </div>
-            </div>
-            <p>새로 나온 영화 재미있대요. 같이 보러 갈 사람?</p>
-            <div class="mt-2 flex justify-between text-gray-500">
-              <button>좋아요</button>
-              <button>댓글</button>
-              <button>공유</button>
-            </div>
-          </div>
-        </div>
-      </main>
-
-      <footer class="bg-gray-200 p-4 text-center">
-        <p>&copy; 2024 항해플러스. All rights reserved.</p>
-      </footer>
-    </div>
-  </div>
-`;
+import { Main, LoginMain } from '../components/Main';
+import Profile from '../components/Profile';
+import Login from '../components/Login';
+import Error from '../components/Error';
+import Router from './router';
+import User from '../store/User';
+
+//renderMainPage
+const user = User();
+const router = Router();
+router.addRoute('/', renderMainPage);
+router.addRoute('/login', renderLoginPage);
+router.addRoute('/profile', renderProfilePage);
+router.addRoute('/404', renderErrorPage);
+
+// 페이지가 로드될 때 초기 페이지 렌더링
+renderMainPage();
+
+document.querySelector('nav').addEventListener('click', (e) => {
+  if (e.target.tagName === 'A') {
+    e.preventDefault();
+    router.navigateTo(e.target.pathname);
+  }
+});
+
+window.addEventListener('DOMContentLoaded', () => {
+  router.navigateTo(window.location.pathname);
+});
+
+function render(component) {
+  document.querySelector('#root').innerHTML = component();
+}
+
+function renderMainPage() {
+  if (user.getUser()) {
+    render(LoginMain);
+    const logoutButton = document.querySelector('a[id="logout"]');
+    logoutButton.addEventListener('click', (e) => {
+      e.preventDefault();
+      user.deleteUser();
+      router.navigateTo('/login');
+    });
+  } else {
+    render(Main);
+  }
+}
+
+function renderProfilePage() {
+  if (!user.getUser()) {
+    router.navigateTo('/login');
+    return;
+  }
+
+  render(Profile);
+
+  if (user.getUser().username) {
+    document.querySelector('input[id="username"]').value =
+      user.getUser().username;
+  }
+  if (user.getUser().email) {
+    document.querySelector('input[id="email"]').value = user.getUser().email;
+  }
+  if (user.getUser().bio) {
+    document.querySelector('textArea[id="bio"]').value = user.getUser().bio;
+  }
+
+  const logoutButton = document.querySelector('a[id="logout"]');
+  logoutButton.addEventListener('click', (e) => {
+    e.preventDefault();
+    user.deleteUser();
+    router.navigateTo('/login');
+  });
+
+  const profileForm = document.getElementById('profile-form');
+
+  const update = (username, email = '', bio = '') => {
```

Comment:
사용하지 않는 파라미터는 가독성과 유지보수성을 위해 지우시는건 어떠신가요? 😆

```suggestion
  const update = () => {
```